nixos/modules/services/web-servers/caddy/default.nix

```diff
@@ -53,17 +53,20 @@ let
           ${concatMapStringsSep "\n" mkVHostConf virtualHosts}
         '';
 
+        nativeCaddy = lib.getExe pkgs.buildPackages.caddy;
```

Comment:
Is `nativeCaddy` necessary, given that `caddy` is part of `nativeBuildInputs`? I'd expect splicing to work correctly here.

Reply:
that's  a good question but in this case why was the if statement then there in the first place ...

... anyway I will just remove the native input as i think this is more clear as the old syntax

Reply:
@ambroisie c7a6fcb3a25dc4e224f10b0f35a4dd98f2223cc3

PS: the native inputs use `cfg.package` not `pkgs.buildPackages.caddy` - so if you build for arm on x86 it would fail ...

Reply:
> PS: the native inputs use cfg.package not pkgs.buildPackages.caddy - so if you build for arm on x86 it would fail ...

That's what I'm getting at: I thought splicing would take care of "switching" to `buildPackages` automatically for `nativeBuildInputs`.

Reply:
https://nixos.org/manual/nixpkgs/stable/#var-stdenv-nativeBuildInputs

so yes the native build inputs might have already worked ... still I personally like to have it explicit rather than more implicite.

I consider the current aproach more readable - but @ambroisie if you have a stron opinion i would go with that else I would say the current state is good to go :)

Reply:
:shrug: will let the maintainers of the module decide.

I can technically see an issue if, say, `cfg.package` is newer than `buildPackages.caddy` and doesn't understand a piece of configuration than `cfg.package` _would_ support.

Not sure if that's an issue that would actually happen though.

Reply:
well that's an good argument i would say ... in this case nativeBuildInputs it is!!!

I still would remove the if statement at the end as we do use nativeBuildInputs ...

PS: who is the package maintainer, can not determain that :/ (its neither in `ci/OWNERS` nor defined in the module)

Reply:
I'd look for previous commiters and reviewers.